compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1574,7 +1574,9 @@ class Typer extends Namer
         .withType(dummy.termRef)
       if (!cls.is(AbstractOrTrait) && !ctx.isAfterTyper)
         checkRealizableBounds(cls, cdef.namePos)
-      if (cls.is(Case) && cls.derivesFrom(defn.EnumClass)) checkEnum(cdef, cls)
+      val firstParent = parents1.head.tpe.dealias.typeSymbol
```

Comment:
This could be moved in the `if` statement

Reply:
I preferred to give it a name anyway, but it’s also used twice.

Reply:
I only see it used once here. I did't mean to inline the RHS but to only evaluate in the then branch:
```scala
if (cls.derivesFrom(defn.EnumClass)) {
  val firstParent = parents1.head.tpe.dealias.typeSymbol
  checkEnum(cdef, cls, firstParent)
}
```